lib/ansible/galaxy/collection.py

```diff
@@ -431,8 +495,15 @@ def from_path(b_path, force, parent=None):
 
         files = info.get('files_file', {}).get('files', {})
 
+        if fallback_metadata and not CollectionRequirement.artifact_info(b_path):
```

Comment:
It's been a while since I last looked at this code. From my understanding `from_path` was used to identify collections which is why we unconditionally set `skip=True` so we never tried to install it again.

Am I understanding the code right where now `from_path` can be called against a cloned repo and that's why we may want to install it (not skip). If that's the case it seems like we may want to have the caller decide whether it's something to skip or not instead of using the is SCM like logic that's currently here.

Reply:
Looking for a better way of doing that.